lessons/misc/intro-bash/lesson.md

```diff
@@ -0,0 +1,354 @@
+
+```
+man ls
+```
+
+## Working with directories
+
+Back to our folder here. While peeking at the contents of a directory is certainly useful, we'll often find ourselves wanting to switch our current directory. This is done using the `cd` (**c**hange **d**irectory) command. You should have downloaded a folder call `intro-bash-demo` and saved it somewhere practical (e.g. your Desktop or home folder) - let's use `cd` to navigate to it. I've a included a snippet of some useful `cd` commands below:
+
+* `cd <directory name>`: Change ito directory. Relative to your current working directory.
+    * `cd intro-bash-demo/`: Change into `intro-bash-demo directory`
+    * `cd ../`: Go back one directory from the previous one. Use `../../` to go back 2 directories.
+    * `cd ~/`: Change to your home directory
+
+**Tip:** You can press TAB to atuocomplete directory and filenames. If a file in the working directory matches the start of the file or directory you've started typing, the shell with autocomplete its name.
+
+Once you've navigated to the `intro-bash-demo` folder, *list* the fles in the folder to view its contents.
+
+You notice some text files (e.g. animals.txt) and directories. We'll return to these shortly.
+
+We can create directories with the `mkdir` command. Let's create a directory called `text-files`.
+
+```
+mkdir text-files
+```
+
+Similarly, you can remove directories with the `rmdir` command. Let's remove the `folder-to-delete` folder.
+
+```
+rmdir folder-to-delete
+```
+
+Hmmm...that didn't work. It turns out `rmdir` only works on empty directories. Let's make sure we don't actually need the files in that directory before trying again to delete it. You can pass directory names to `ls` to its contents.
+
+```
+ls folder-to-delete
+```
+
+Now that we're sure we don't need that folder anymore, let's **f**orcefully and **r**ecursively delete it using the `rm` (for *remove*) command. Note, this command will not warn you and the content will be gone forever.
+
+```
+rm -rf folder-to-delete
+```
+
+`rm` can also be used on files without the `-rf` flags. Try remove `to-delete.txt`
+
+## Working with files
+
+Beyond navigating directories, bash offers a suite of useful tools to do all sorts of things with files themselves.
+
+Let's examine `hello_world.py` in greater detail. We can do this with `cat`, which simply spits out the contents of a file to the terminal's output.
+
+```
+cat hello_world.py
+```
+
+When it comes to looking at individual files, however, `cat` is generally not very practical for anything but tiny files like this one. Generally, we would want to use `less`, a program known as a *pager* that is designed to let users scroll through files (no matter how large) by only generating a single 'page worth' at a time.
+
+Let's use `less` to view the contents of the larger file, `animals.txt`.
+
+```
+less animals.txt
```

Comment:
It might be helpful to add a few things about `less` itself in here - ie using arrow keys/vim bindings to navigate, knowing that bash is only rendering what's on the screen (which means that extremely large files can be opened at little memory cost), and - most importantly - that we have to press `q` to exit!